### ISSUE TITLE
Add support for the Raspberry Pi (ARM)

### ISSUE THREAD

**Issue by MakuSensei** [AnarchyLinux/installer#645]
I would like anarchy-creator.sh to create a custom version for raspi3

**Comment by ljoli** [comment]
Humm, this is the same issue of #53.
no ?

**Comment by InvalidSudo** [comment]
It's not nearly as simple as porting x86_64 Anarchy Linux for i386.

**Comment by MarcusE1W** [comment]
I would be interested as well. Does someone know, what are the main things that would need to be adapted for the ARM Archlinux ?

**Comment by JustMeErazem** [comment]
@MarcusE1W Can't we just reuse the [Arch Linux ARM](https://archlinuxarm.org/) project?

**Comment by TuxNix** [comment]
It would be nice to have a script just for configuration. Raspbian has a nice script.
We could adopt it to Arch Linux ARM and put it to the AUR.

**Comment by JustMeErazem** [comment]
@TuxNix Which script are you talking about? Do you mean `raspi-config`?

**Comment by TuxNix** [comment]
Yes, I mean raspi-config. A script with those features would be good to start with.
Later it would be nice to have GUI-Installaton, nextcloud server and so on.
Following step could be to have an image, just with ARCH LINUX ARM base and that script.
Raspberry Pi 4 is there and I guess, a lot of people would like to have a comfortable way to run ARCH on it.

**Comment by JustMeErazem** [comment]
That's an interesting idea. I'd love to help, but I'm not really sure where to start. If you are interested in creating the script, your contributions would be very helpful. How do you think we should go about including it in Anarchy though? Because to me this looks more like a seperate project, which could later be incorporated into Anarchy. You could create a seperate branch with only the script and work on it there. What do you think about this?

**Comment by TuxNix** [comment]
Sounds nice to me. But actually I have no time to do this. I'll tell you when I'll start working on the script.

**Comment by PandaFoss** [comment]
I have a Raspberry Pi 3 with Arch Linux ARM. I don't know what functions the raspi-config script brings, but I find it interesting that Anarchy Linux also incorporates this platform. I will think about it...

**Comment by PandaFoss** [comment]
Well, I found [this thread](https://archlinuxarm.org/forum/viewtopic.php?f=4&t=12533) in the Arch Linux ARM forum that talks about a version of the raspi-config script for Arch. The [repository](https://github.com/jgmdev/alarm-raspi-config) is somewhat old and outdated, but I think it is preferable to start from scratch. Perhaps from the iso-generator script an ISO image can be generated with a raspi-config-like script to facilitate some Arch Linux configurations in the RPi.

**Comment by JustMeErazem** [comment]
Yup, we could make a few new functions for the Pi and reuse the ones that copy the files etc.

But we have to check if all the packages (especially the ones installed from AUR) have ARM support.

**Comment by PandaFoss** [comment]
@JustMeErazem Yes, but I don't promise to have anything ready for the next version. However, I would love to work on it.

**Comment by JustMeErazem** [comment]
No worries, there are no deadlines ;)

**Comment by PandaFoss** [comment]
## Issue update
I am working on this new functionality. I was thinking and the most convenient thing will be to add it to the usual ISO as an optional tool before starting the installer (instead of redesigning the installer to support the armv7h architecture). That is, once Anarchy is started, `anarchy --rpi` can be run to install the operating system on an SD card or on a USB stick, and then you can continue using Live to start the classic Anarchy installer or anything you want to do (for example, I used Anarchy on a pendrive to fix errors at startup).
The script in question is almost functional, but when I solve some remaining details, I will upload it to my account so they can test it before it is incorporated into Anarchy Linux.
Later I will add a script within that installation that simulates the functionality of `raspi-config` script.
### Let's continue developing!

**Comment by JustMeErazem** [comment]
Really cool idea, but this would mean you'd need to flash Anarchy on a usb and run it off of your pc to install it to a seperate sd card.

The problem with devices like RPi-s (and others) is that you have to install the os to the sdcard directly, and then immediately start using it (there is no installation process). This is why projects like ArchLinuxARM and Raspbian have complete system images.

I think the correct solution to this problem would be to add an '--rpi' option to the installer which would use the ArchLinuxARM image as the base and work just like the normal Anarch installer, but without the partitioning (since it will already be installed to the system).

I want to hear your opinion on this though, this is just my proposal.

**Comment by PandaFoss** [comment]
Your proposal is interesting. I think it is a better alternative to the one I proposed, although it involves a lot of work to edit the installer and support the new architecture and the new installation method (Arch Linux ARM has a tarball instead of a traditional ISO image) . However, I will work on it as soon as possible!

**Comment by condor2** [comment]
You could fork this and create for Raspberry.
Or just create a raspberry-iso-generator....

**Comment by PandaFoss** [comment]
I will consider your suggestions, although I have already encountered some inconveniences. First, there is no ISO image of Arch Linux ARM that can be booted, or that can be taken as the basis for creating the Anarchy image. It remains for me to investigate whether it is possible to create an image for an architecture different from that of my host computer (x86_64 instead of armv7h). Second, I thought about creating the Anarchy ISO image normally (64-bit) and running `chroot` inside the Arch Linux ARM installation to execute the necessary changes, but I also found it impossible to run `chroot` inside an installation of another architecture (sample input/output error).
My English is not very good but I think what I say is understood.
Keep investigating!
